Not Compatible w/ Python 3.14

| | |
|:---|:---|
|**_F´ Version_**| v4.1.0 |
|**_Affected Component_**|  fprime-util |
---
## Problem Description

Attempted to run the led-blinker tutorial with python 3.14. When running the command `fprime-bootstrap project` it would complete with some build issues but the output was indicative of a success so I continued (see output below). Then I activated the virtual environment and attempted to run `fprime-util generate` and that gave an error `-bash: fprime-util: command not found`. After talking to Michael he said the install went wrong and had me attempt to instal the reuirements.txt. This resulted in the same build errors as the output below. He then suggested I uninstall python 3.14 and try python 3.13 which did work. 

``` 
CMake Error at /var/folders/77/q_ryk3kx2jvd708d4n6rl53r0000gq/T/tmp5d4z1cos/build/_deps/bundled_libzmq-src/CMakeLists.txt:5 (cmake_minimum_required):
        Compatibility with CMake < 3.5 has been removed from CMake.
      
        Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
        to tell CMake that the project requires at least <min> but has been updated
        to work with policies introduced by <max> or earlier.
      
        Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
      
      
      -- Configuring incomplete, errors occurred!
      
      *** CMake configuration failed
      [end of output]
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pyzmq
Failed to build pydantic_core pyzmq
error: failed-wheel-build-for-install

× Failed to build installable wheels for some pyproject.toml based projects
╰─> pydantic_core, pyzmq

################################################################

Congratulations! You have successfully created a new F´ project.

A git repository has been initialized and F´ has been added as a
submodule, you can now create your first commit.

Get started with your F´ project:

-- Remember to always activate the virtual environment --
cd led-blinker
. fprime-venv/bin/activate

-- Create a new component --
fprime-util new --component

-- Create a new deployment --
fprime-util new --deployment

################################################################

```

## Context / Environment

Execute `fprime-util version-check` and share the output.

Writing this ticket after I switched python versions so I don't have an output for this step.

## How to Reproduce

1. Attempt to make a project with python 3.14 

## Expected Behavior

Fail to `fprime-util generate`

--- a/.github/workflows/ext-raspberry-led-blinker.yml
+++ b/.github/workflows/ext-raspberry-led-blinker.yml
@@ -5,15 +5,15 @@ name: "External Repo: RPI LedBlinker"
 
 on:
   push:
-    branches: [ devel, release/** ]
+    branches: [devel, release/**]
   pull_request:
     # The branches below must be a subset of the branches above
-    branches: [ devel, release/** ]
+    branches: [devel, release/**]
     paths-ignore:
-      - 'docs/**'
-      - '**.md'
-      - '.github/actions/spelling/**'
-      - '.github/ISSUE_TEMPLATE/**'
+      - "docs/**"
+      - "**.md"
+      - ".github/actions/spelling/**"
+      - ".github/ISSUE_TEMPLATE/**"
 # Cancel in-progress runs if a newer run is started on a given PR
 concurrency:
   group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
@@ -85,6 +85,7 @@ jobs:
         run: |
           python -m venv venv
           . venv/bin/activate
+          pip install --upgrade pip
           pip install -r requirements.txt
       - name: "Artifacts Download"
         uses: actions/download-artifact@v4

--- a/.github/workflows/pip-check.yml
+++ b/.github/workflows/pip-check.yml
@@ -20,7 +20,7 @@ jobs:
     runs-on: ${{ matrix.runner }}
     strategy:
       matrix:
-        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]
+        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13", "3.14"]
         # Purposefully test on both ARM and Intel macOS (macos-latest is ARM)
         runner: [macos-15-intel, macos-latest, ubuntu-22.04, ubuntu-latest]
     steps:

--- a/requirements.txt
+++ b/requirements.txt
@@ -13,7 +13,7 @@ click==8.1.7
 cmake==3.26.0
 colorlog==6.8.2
 cookiecutter==2.6.0
-CT3==3.3.3.post1
+CT3==3.4.0.post5
 et-xmlfile==1.1.0
 Flask==3.0.3
 Flask-Compress==1.15
@@ -34,7 +34,7 @@ legacy-cgi==2.6.1; python_version >= "3.13"
 lxml==6.0.2
 Markdown==3.7
 markdown-it-py==3.0.0
-MarkupSafe==2.1.5
+MarkupSafe==3.0.3
 mdurl==0.1.2
 ninja==1.11.1.4
 openpyxl==3.1.5
@@ -50,8 +50,8 @@ pytest==8.3.3
 python-dateutil==2.9.0.post0
 python-slugify==8.0.4
 pytz==2024.2
-PyYAML==6.0.2
-pyzmq==26.2.0
+PyYAML==6.0.3
+pyzmq==27.1.0
 requests==2.32.4
 rich==13.9.2
 setuptools==78.1.1
@@ -63,5 +63,5 @@ typing-extensions==4.15.0
 urllib3==2.6.3
 Werkzeug==3.1.5
 zipp==3.19.1
-zstandard==0.23.0
+zstandard==0.24.0
 pip

--- a/requirements.txt
+++ b/requirements.txt
@@ -42,8 +42,8 @@ packaging==24.1
 pexpect==4.9.0
 pluggy==1.5.0
 ptyprocess==0.7.0
-pydantic==2.9.2
-pydantic_core==2.23.4
+pydantic==2.12.5
+pydantic_core==2.41.5
 Pygments==2.18.0
 pyserial==3.5
 pytest==8.3.3
@@ -59,7 +59,7 @@ six==1.16.0
 text-unidecode==1.3
 tomli==2.0.1
 types-python-dateutil==2.9.0.20241003
-typing-extensions==4.12.2
+typing-extensions==4.15.0
 urllib3==2.6.0
 Werkzeug==3.1.5
 zipp==3.19.1

--- a/requirements.txt
+++ b/requirements.txt
@@ -31,7 +31,7 @@ iniconfig==2.0.0
 itsdangerous==2.2.0
 Jinja2==3.1.6
 legacy-cgi==2.6.1; python_version >= "3.13"
-lxml==5.3.0
+lxml==6.0.2
 Markdown==3.7
 markdown-it-py==3.0.0
 MarkupSafe==2.1.5

